modules/rostests/winetests/wtsapi32/wtsapi.c

```diff
@@ -16,92 +16,456 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
  */
 
+#include <ntstatus.h>
+#define WIN32_NO_STATUS
 #include <stdarg.h>
 #include <stdlib.h>
 #include <windef.h>
 #include <winbase.h>
 #include <wine/winternl.h>
+#include <lmcons.h>
 #include <wtsapi32.h>
+#define PSAPI_VERSION 1
+#include <psapi.h>
 
 #include "wine/test.h"
 
-static void test_WTSEnumerateProcessesW(void)
+static BOOL (WINAPI *pWTSEnumerateProcessesExW)(HANDLE server, DWORD *level, DWORD session, WCHAR **info, DWORD *count);
+static BOOL (WINAPI *pWTSFreeMemoryExW)(WTS_TYPE_CLASS class, void *memory, ULONG count);
+
+static const SYSTEM_PROCESS_INFORMATION *find_nt_process_info(const SYSTEM_PROCESS_INFORMATION *head, DWORD pid)
 {
-    BOOL found = FALSE, ret;
-    DWORD count, i;
-    PWTS_PROCESS_INFOW info;
-    WCHAR *pname, nameW[MAX_PATH];
+    for (;;)
+    {
+        if ((DWORD)(DWORD_PTR)head->UniqueProcessId == pid)
+            return head;
+        if (!head->NextEntryOffset)
+            break;
+        head = (SYSTEM_PROCESS_INFORMATION *)((char *)head + head->NextEntryOffset);
+    }
+    return NULL;
+}
+
+static void check_wts_process_info(const WTS_PROCESS_INFOW *info, DWORD count)
+{
+    ULONG nt_length = 1024;
+    SYSTEM_PROCESS_INFORMATION *nt_info = malloc(nt_length);
+    WCHAR process_name[MAX_PATH], *process_filepart;
+    BOOL ret, found = FALSE;
+    NTSTATUS status;
+    DWORD i;
 
-    GetModuleFileNameW(NULL, nameW, MAX_PATH);
-    for (pname = nameW + lstrlenW(nameW); pname > nameW; pname--)
+    GetModuleFileNameW(NULL, process_name, MAX_PATH);
+    process_filepart = wcsrchr(process_name, '\\') + 1;
+
+    while ((status = NtQuerySystemInformation(SystemProcessInformation, nt_info,
+            nt_length, NULL)) == STATUS_INFO_LENGTH_MISMATCH)
+    {
+        nt_length *= 2;
+        nt_info = realloc(nt_info, nt_length);
+    }
+#ifdef __REACTOS__
+    ok(!status, "got %#x\n", status);
+#else
+    ok(!status, "got %#lx\n", status);
+#endif
```

Comment:
```suggestion
    ok(!status, "got %#lx\n", status);
```
(by removing the ROS_LONG64)